### PR TITLE
CRS-279 separate TypingIndicator

### PR DESCRIPTION
**Pull request by mahboubii**
pending https://github.com/GetStream/stream-chat-js/pull/440

### CODE REVIEW

**Comment by mahboubii** [comment]
@jaapbakker88 it probably needs some styling updates too

File: src/components/TypingIndicator/TypingIndicator.js [hunk 1]
```diff
@@ -1,42 +1,40 @@
...
+const TypingIndicator = () => {
+  const { typing, client } = useContext(ChannelContext);
+
+  if (!typing || !client) return null;
+
+  const users = Object.values(typing).filter(
+    ({ user }) => user?.id !== client.user?.id,
+  );
+
+  console.log('users :>> ', users);
```
Comment (vicnicius) [inline src/components/TypingIndicator/TypingIndicator.js:21]: 🔪

Comment (mahboubii) [inline src/components/TypingIndicator/TypingIndicator.js:21]:
```suggestion
```

**Comment by jaapbakker88** [comment]
@mahboubii let me know if you want me to look at the styling for this.

**Review by vicnicius — APPROVED** [review]

File: src/components/MessageList/MessageListInner.js [hunk 2]
```diff
@@ -289,7 +290,7 @@ const MessageListInner = (props) => {
       {...internalInfiniteScrollProps}
     >
       <ul className="str-chat__ul">{elements}</ul>
-
+      {!threadList && <TypingIndicator />}
```
Comment (vicnicius) [inline src/components/MessageList/MessageListInner.js:293]: would be nice to also cover this with some tests.

Comment (mahboubii) [reply src/components/MessageList/MessageListInner.js:293]: agree, we don't have any tests set up for the MessageList yet so out of the scope of this PR

**Review by jaapbakker88 — APPROVED** [review]

**Pull request merged**